include/igl/cat.cpp

```diff
@@ -251,6 +251,63 @@ IGL_INLINE void igl::cat(const std::vector<std::vector< Mat > > & A, Mat & C)
   }
 }
 
+template <typename T, typename DerivedC>
+IGL_INLINE void igl::cat(const int dim, const std::vector<T> & A, Eigen::PlainObjectBase<DerivedC> & C)
+{
+  assert(dim == 1 || dim == 2);
+  using namespace Eigen;
+
+  const int num_mat = A.size();
+  if(num_mat == 0)
+  {
+    C.resize(0,0);
+    return;
+  }
+
+  if(dim == 1)
+  {
+    const int A_cols = A[0].cols();
+
+    int tot_rows = 0;
+    for(const auto & m : A)
+    {
+      tot_rows += m.rows();
+    }
+
+    C.resize(tot_rows, A_cols);
+
+    int cur_row = 0;
+    for(int i = 0; i < num_mat; i++)
+    {
+      assert(A_cols == A[i].cols());
+      C.block(cur_row,0,A[i].rows(),A_cols) = A[i];
+      cur_row += A[i].rows();
+    }
+  }else if(dim == 2)
```

Comment:
And go to newline after this brace.

Reply:
Sorry, thanks for your patience with this.

I guess I don't actually know what the preferred brace-style is completely. I copied this style from earlier in the file. As far as I can tell, the [style guidelines](https://libigl.github.io/style-guidelines/) does not address braces.

Reply:
Hmm you're right. I've been trying to match the already existing codestyle. I think braces on a newline for `for` and `if` are a waste of space, but that's not up to me. I'll see if we can update the guideline accordingly.